src/components/workspace-utils.js

```diff
@@ -340,6 +341,63 @@ export const WorkspaceTagSelect = props => {
   })
 }
 
+export const WorkspaceStarControl = ({ workspace, stars, setStars, style }) => {
+  const [updating, setUpdating] = useState(false)
+
+  const { workspace: { workspaceId } } = workspace
+  const isStarred = _.includes(workspaceId, stars)
+
+  //Thurloe has a limit of 2048 bytes for its VALUE column. That means we can store a max of 55
+  //workspaceIds in list format. We'll use 50 because it's a nice round number and should be plenty
+  //for the intended use case. If we find that 50 is not enough, consider introducing more powerful
+  //workspace organization functionality like folders
+  const MAX_STARRED_WORKSPACES = 50
+  const maxStarredWorkspacesReached = _.size(stars) >= MAX_STARRED_WORKSPACES
+
+  const refreshStarredWorkspacesList = async () => {
+    const { starredWorkspaces } = Utils.kvArrayToObject((await Ajax().User.profile.get()).keyValuePairs)
```

Comment:
It is odd that the get and set of preferences is not symmetrical (`register/profile` vs. `api/profile/preferences`).

Reply:
Agreed- these APIs are pretty ancient and there tends to be a lot of asymmetry and non-RESTful quirks about them